File: frontend/src/api/operations.ts
import { apiClient } from "./client";
import type { OperationalDevice, OperationsEvent, OperationsStatus, RecentMeasurements } from "../types/operations";

export async function getOperationsStatus(): Promise<OperationsStatus> {
  const response = await apiClient.get<OperationsStatus>("/operations/status");
  return response.data;
}

export async function getOperationalDevices(): Promise<OperationalDevice[]> {
  const response = await apiClient.get<OperationalDevice[]>("/operations/devices");
  return response.data;
}

export async function getRecentMeasurements(): Promise<RecentMeasurements> {
  const response = await apiClient.get<RecentMeasurements>("/operations/recent-measurements");
  return response.data;
}

export async function getOperationsEvents(): Promise<OperationsEvent[]> {
  const response = await apiClient.get<OperationsEvent[]>("/operations/events");
  return response.data;
}

File: backend/app/api/operations.py
from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Depends

from backend.app.api.deps import Database, require_permission
from backend.app.services.operations_service import (
    operational_devices,
    operations_status,
    recent_events,
    recent_measurements_summary,
)


router = APIRouter(prefix="/api/operations", tags=["operations"])


@router.get("/status")
def status(
    conn: Database,
    _: object = Depends(require_permission("view_dashboard")),
) -> dict[str, Any]:
    return operations_status(conn)


@router.get("/devices")
def devices(
    conn: Database,
    _: object = Depends(require_permission("view_dashboard")),
) -> list[dict[str, Any]]:
    return operational_devices(conn)


@router.get("/recent-measurements")
def recent_measurements(
    _: object = Depends(require_permission("view_dashboard")),
) -> dict[str, Any]:
    return recent_measurements_summary()


@router.get("/events")
def events(
    conn: Database,
    _: object = Depends(require_permission("view_dashboard")),
) -> list[dict[str, Any]]:
    return recent_events(conn)
